homeassistant/components/teslemetry/climate.py

```diff
@@ -0,0 +1,130 @@
+    data = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        TeslemetryClimateEntity(vehicle, TeslemetryClimateSide.DRIVER)
+        for vehicle in data
+    )
+
+
+class TeslemetryClimateEntity(TeslemetryVehicleEntity, ClimateEntity):
+    """Vehicle Location Climate Class."""
+
+    _attr_precision = PRECISION_HALVES
+    _attr_min_temp = 15
+    _attr_max_temp = 28
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_modes = [HVACMode.HEAT_COOL, HVACMode.OFF]
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_preset_modes = ["off", "keep", "dog", "camp"]
+
+    @property
+    def hvac_mode(self) -> HVACMode | None:
+        """Return hvac operation ie. heat, cool mode."""
+        if self.get("climate_state_is_climate_on"):
+            return HVACMode.HEAT_COOL
+        return HVACMode.OFF
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        return self.get("climate_state_inside_temp")
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the temperature we try to reach."""
+        return self.get(f"climate_state_{self.key}_setting")
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return self.get("climate_state_max_avail_temp", self._attr_max_temp)
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return self.get("climate_state_min_avail_temp", self._attr_min_temp)
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode."""
+        return self.get("climate_state_climate_keeper_mode")
+
+    async def async_turn_on(self) -> None:
+        """Set the climate state to on."""
+        with handle_command():
+            await self.wake_up_if_asleep()
+            await self.api.auto_conditioning_start()
+        self.set(("climate_state_is_climate_on", True))
+
+    async def async_turn_off(self) -> None:
+        """Set the climate state to off."""
+        with handle_command():
+            await self.wake_up_if_asleep()
+            await self.api.auto_conditioning_stop()
+        self.set(
+            ("climate_state_is_climate_on", False),
+            ("climate_state_climate_keeper_mode", "off"),
+        )
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set the climate temperature."""
+        temp = kwargs[ATTR_TEMPERATURE]
+        with handle_command():
+            await self.wake_up_if_asleep()
+            await self.api.set_temps(
+                driver_temp=temp,
+                passenger_temp=temp,
+            )
+
+        self.set((f"climate_state_{self.key}_setting", temp))
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
```

Comment:
Do we need HVAC when we have turn on and turn off?

Reply:
Yes. Otherwise:
```def set_hvac_mode(self, hvac_mode: HVACMode) -> None:
        """Set new target hvac mode."""
        raise NotImplementedError()```